backend/schema/schema_test.go

```diff
@@ -54,16 +54,16 @@ module todo {
 
 module foo {
   // A comment
-  enum Color(String) {
-	Red("Red")
-	Blue("Blue")
-	Green("Green")
+  enum Color {
+	Red String = "Red"
+	Blue String = "Blue"
+	Green String = "Green"
   }
```

Comment:
We didn't want to go with Swift-style?

```swift
enum Color: String {
  Red = "Red"
  Blue = "Blue"
  Green = "Green"
}
```

Reply:
Ah it's probably simpler to have a single form as you do now?

Reply:
yeah, the rationale was that because of runtime constraints FTL enums have the following requirements/limitations (unlike Swift enums):
1. an enum can be either a "value" enum or a "type" enum
2. if an enum is a "value" enum, each variant must have a corresponding constant/"raw" value and all values must share a single type
3. if an enum is a "type" enum, no variants can have a "raw" value

so we wanted to represent enums of different varieties with consistency. but here's a secondary thought/proposal—

"value" enum schema string:
```
enum Color: String {
  Red = "Red"
  Blue = "Blue"
  Green = "Green"
}
```

"type" enum sch string:
```
enum Color {
  Red  String
  Blue Int
}
```
or
```
 // i don't know if there's really a use-case for something like this...
enum Color: String {
   Red
   Blue
}
```

we keep "Type" an outermost attribute of the `schema.Enum`, but it is optional and only present if all variants share a single type. 

we _do not_ treat "Value" as optional in the variants—every variant must have a value. But we introduce `typevalue.go` (like `intvalue.go/stringvalue.go`), and in "type" enums, the value will be the type itself.

wdyt @alecthomas @deniseli?